### PR TITLE
#167377554 Search functionality for articles

### PR DESCRIPTION
**Pull request by Nennyfills**
### What does this PR do?
This pull request enables all user to search for articles.

#### Description of Task to be completed
-  Create RESTful route to get an article by category, title, author, and keyword
-  Add validation to articles category, title, author, and keyword queries
-  Write a test to test articles validation and route

#### How should this be manually tested?
- git clone this branch
- run `npm install`
- Using postman: 
    Get  - `http://localhost:{port}/api/v1/articles?author`

#### Any Background Context You want to provide?
N/A
#### What are the relevant pivotal tracker stories?

[#167377554](https://pivotaltracker.com/story/show/167377554)

#### Screenshots (if appropriate)
Screenshot 2019-08-07 at 1 29 53 PM
Screenshot 2019-08-07 at 1 28 42 PM
Screenshot 2019-08-07 at 1 28 13 PM
Screenshot 2019-08-07 at 1 27 44 PM
Screenshot 2019-08-07 at 1 27 31 PM
Screenshot 2019-08-07 at 1 26 55 PM

#### Questions:
N/A

### CODE REVIEW

File: src/helpers/genericArticleHelper.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+const extractArticles = (result) => {
+  const data = [];
+
+  result.forEach((entry) => {
+    const summary = entry['article.summary'];
+    const body = entry['article.body'];
+    const suspended = entry['article.suspended'];
+    const status = entry['article.status'];
+    const coverImageUrl = entry['article.coverImageUrl'];
+    const followUpId = entry['article.title'];
```
Comment (Merkll) [inline src/helpers/genericArticleHelper.js]: let's use ```Array.map``` in place of ```forEach```

File: src/helpers/paginator.js [hunk 2]
```diff
@@ -7,6 +7,7 @@ import { NotFoundError } from './errors';
  *
  * @param {Object} Source - Model object
  * @param {Object} options
+//  * @param {query} query - For model query
```
Comment (Merkll) [inline src/helpers/paginator.js]: Check and Remove all comments

File: src/helpers/genericArticleHelper.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+const filter = (title, tag, author, keyword) => {
+  const queryParams = {
+    query: {
+      '$article.title$': { [Op.iLike]: `%${title}%` },
+      '$tag.category$': { [Op.iLike]: `${tag}` },
+    },
+    author: {
+      [Op.or]: {
+        '$author.firstName$': { [Op.iLike]: `%${author}%` },
+        '$author.lastName$': { [Op.iLike]: `%${author}%` },
```
Comment (Merkll) [inline src/helpers/genericArticleHelper.js:28]: Good Job searching in all fields

**Review by Merkll — APPROVED** [review]

File: src/helpers/paginator.js (hunk 2 above)
Comment (Nennyfills) [reply src/helpers/paginator.js]: Done.

File: src/helpers/genericArticleHelper.js (hunk 3 above)
Comment (Nennyfills) [reply src/helpers/genericArticleHelper.js:28]: Thanks

File: src/helpers/genericArticleHelper.js (hunk 1 above)
Comment (Nennyfills) [reply src/helpers/genericArticleHelper.js]: Done.

File: src/helpers/paginator.js (hunk 2 above)
Comment (Nennyfills) [reply src/helpers/paginator.js]: Done

File: src/helpers/genericArticleHelper.js (hunk 3 above)
Comment (Nennyfills) [reply src/helpers/genericArticleHelper.js]: Thanks.

**Pull request merged**